src/circuit/listener.js

```diff
@@ -20,8 +20,8 @@ module.exports = (libp2p) => {
     const deleted = listeningAddrs.delete(connection.remotePeer.toB58String())
 
     if (deleted) {
-      // TODO push announce multiaddrs update
-      // libp2p.identifyService.pushToPeerStore()
+      // Announce listen addresses change
+      listener.emit('listening')
```

Comment:
Not a fan of using the `listening` event here. But, the other option is to have a new event for the not listening or trigger a manual update of the peerRecord/addressBook self. I was trying to be compliant with the `interface-transport`. 
What do you think @jacobheun ?

Reply:
I guess I can use the `close` event from the listeners and have a `transportManager.emit('change')`. But, circuit would need to emit close here and transportManager would track `close` and `listening`